spec/lib/whitehall_importer/embed_body_references_spec.rb

```diff
@@ -12,49 +12,106 @@
-        images: ["foo.png"],
+    it "ignore Whitehall attachment bang embeds that are neither start of the string or preceeded by new lines" do
+      body = described_class.call(
+        body: "!@1 test !@2",
+        attachments: ["file.pdf", "file.csv"],
       )
 
-      expect(govspeak_body).to eq("Bar  Baz")
+      expect(body).to eq("[Attachment:file.pdf] test !@2")
     end
 
-    it "converts Whitehall attachment syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "!@1 test !@2",
-        attachments: ["file.pdf", "download.csv"],
+    it "ignore any image markdown that doesn't resolve to an image" do
+      body = described_class.call(
+        body: "Bar !!2 Baz",
+        images: ["file.jpg"],
       )
 
-      expect(govspeak_body).to eq("[Attachment:file.pdf] test [Attachment:download.csv]")
+      expect(body).to eq("Bar !!2 Baz")
     end
 
-    it "converts Whitehall inline attachment syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "[InlineAttachment:1] test [InlineAttachment:2]",
-        attachments: ["file.pdf", "download.csv"],
+    it "ignore any attachment markdown that doesn't resolve to an attachment" do
+      body = described_class.call(
+        body: "Bar !@2 Baz",
+        attachments: ["file.pdf"],
       )
 
-      expect(govspeak_body).to eq("[AttachmentLink:file.pdf] test [AttachmentLink:download.csv]")
+      expect(body).to eq("Bar !@2 Baz")
+    end
+
+    it "converts image and attachment embeds prefixed by single new lines to be prefixed by two" do
+      prefix_conversions = [
+        { prefix: "\n", conversion: "\n\n" },
+        { prefix: "\r\n", conversion: "\r\n\r\n" },
+      ]
+
+      prefix_conversions.each do |prefix_conversion|
+        prefixed = []
+        prefixed << "#{prefix_conversion[:prefix]}!!1"
+        prefixed << "#{prefix_conversion[:prefix]}!@1"
+
+        converted = []
+        converted << "#{prefix_conversion[:conversion]}[Image:file.jpg]"
+        converted << "#{prefix_conversion[:conversion]}[Attachment:file.pdf]"
+
+        body = described_class.call(
+          body: prefixed.join,
+          images: ["file.jpg"],
+          attachments: ["file.pdf"],
+        )
+
+        expect(body).to eq(converted.join)
+      end
     end
 
-    it "removes any attachment markdown that doesn't resolve to an attachment" do
+    it "doesn't change the prefixes of images and attachments preceeded by two or more new lines" do
+      prefixes = [
+        "\n\n", "\r\n\r\n",
+        "\n\n\n", "\r\n\r\n\r\n",
+        "\n\n\n\n", "\r\n\r\n\r\n\r\n"
+      ]
+
+      prefixes.each do |prefix|
+        body = described_class.call(
+          body: "#{prefix}!!1#{prefix}!@1",
+          images: ["file.jpg"],
+          attachments: ["file.pdf"],
+        )
+
+        expect(body)
+          .to eq("#{prefix}[Image:file.jpg]#{prefix}[Attachment:file.pdf]")
+      end
```

Comment:
Cool this works - looped expectations are frowned upon in tests but can see the value here.

Reply:
I think there are - rare - occasions when this does work.  I feel this is one of those occasions too.